File: src/LeetCode/TreeNode.java
package LeetCode;

public class TreeNode {
    int val;
    public TreeNode left;
    public TreeNode right;
    public TreeNode() {}
    public TreeNode(int val) { this.val = val; }
    public TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }

    public void print() {
        System.out.println(" " + val);
        if (left != null) {
            left.print();;
        } else {
            System.out.println(" null");
        }
        if (right != null) {
            right.print();;
        } else {
            System.out.println(" null");
        }
    }
}

File: LeetCode/Basic/TreeNode.swift
//
//  TreeNode.swift
//  LeetCode
//
//  Created by stone on 2022/3/21.
//

import Foundation

public class TreeNode {
    public var val: Int
    public var left: TreeNode?
    public var right: TreeNode?
    public init() { self.val = 0; self.left = nil; self.right = nil; }
    public init(_ val: Int) { self.val = val; self.left = nil; self.right = nil; }
    public init(_ val: Int, _ left: TreeNode?, _ right: TreeNode?) {
        self.val = val
        self.left = left
        self.right = right
    }
    
    public convenience init?(_ tree: [Int?], level: Int, index: Int) {
        self.init()
        
        if tree.count < (2 << (level - 1) + index) {
            return nil
        }
    }
}
